controllers/clustergroupupgrade_controller.go

```diff
@@ -113,13 +114,37 @@ func (r *ClusterGroupUpgradeReconciler) Reconcile(ctx context.Context, req ctrl.
 					return ctrl.Result{}, err
 				}
 
+				// Set default values for status reason and message.
 				var statusReason, statusMessage string
 				statusReason = "UpgradeNotStarted"
 				statusMessage = "The ClusterGroupUpgrade CR is not enabled"
-				if clusterGroupUpgrade.Spec.Enable == true {
-					statusReason = "UpgradeNotCompleted"
-					statusMessage = "The ClusterGroupUpgrade CR has upgrade policies that are still non compliant"
-					clusterGroupUpgrade.Status.Status.StartedAt = metav1.Now()
+				requeueAfter := 5 * time.Minute
+
+				// If the remediation plan is empty, update the status.
+				if clusterGroupUpgrade.Status.RemediationPlan == nil {
+					statusReason = "UpgradeCannotStart"
+					statusMessage = "The ClusterGroupUpgrade CR has no NonCompliant clusters for the specified managed policies"
+				} else if clusterGroupUpgrade.Spec.Enable == true {
+					// Check if there are any CRs running that are blocking the start of the current one.
+					blockingCRs, err := r.blockingCRsRunning(ctx, clusterGroupUpgrade)
```

Comment:
Minor nit on naming. A blocking CR that is pending (not yet running) should still block this CR. See note in blockingCRsRunning below.